File: typescript/grains/grains.ts
export default class Grains {

    static totalGrains: number = 0

    static square(square: number): number {
        if (square <= 0 || square > 64) { throw Error }

        let grains = 1
        let count = 1
        this.totalGrains += grains

        while (count !== square) {
            grains = grains * 2
            count++
            this.totalGrains += grains
        }

        return grains
    }

    static total(): number { // failing
        return (2 ** 64) - 1
    }
}

File: csharp/grains/Grains.cs
using System;
using System.Numerics;

public static class Grains
{
    public static BigInteger Square(int n) => (n < 1 || n > 64) ? throw new ArgumentOutOfRangeException() : BigInteger.Pow(2, n - 1);
    public static BigInteger Total() => BigInteger.Pow(2, 64) - 1;
}
